compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1439,17 +1439,32 @@ mod signal_handler {
 
     extern "C" fn print_stack_trace(_: libc::c_int) {
         const MAX_FRAMES: usize = 256;
+        // Reserve data segment so we don't have to malloc in a signal handler, which might fail
         static mut STACK_TRACE: [*mut libc::c_void; MAX_FRAMES] =
             [std::ptr::null_mut(); MAX_FRAMES];
         unsafe {
+            // Collect return addresses
             let depth = libc::backtrace(STACK_TRACE.as_mut_ptr(), MAX_FRAMES as i32);
             if depth == 0 {
                 return;
             }
-            backtrace_symbols_fd(STACK_TRACE.as_ptr(), depth, 2);
+            // Just a stack trace is cryptic. Explain what we're doing.
+            write_raw_err("error: rustc interrupted by SIGSEGV, printing stack trace:\n\n");
+            // Elaborate return addrs into symbols and write them directly to stderr
+            backtrace_symbols_fd(STACK_TRACE.as_ptr(), depth, libc::STDERR_FILENO);
```

Comment:
can we print a message here reminding people that this could be a stackoverflow and there's a backtrace above? you don't immediately see the message above and when you see "oh no rustc segfaulted what the hell went wrong??" you're immediately assured by "don't worry it's probably just a stackoverflow". Rust users (like me) are not used to segfaults from stackoverflows (thanks, std^^).